homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -153,6 +172,30 @@ def off_delay_listener(now):
         def state_message_received(msg):
             """Handle a new received MQTT state message."""
             payload = msg.payload
+            # auto-expire enabled?
+            expire_after = self._config.get(CONF_EXPIRE_AFTER)
+            self._available = True
```

Comment:
Same comment, we should not mess with availability here, let the mixin handle it so it beahaves the same for all MQTT platforms.